haystack_experimental/components/generators/chat/bedrock.py

```diff
@@ -0,0 +1,148 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import base64
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from haystack import component
+from haystack.dataclasses.streaming_chunk import StreamingCallbackT
+from haystack.lazy_imports import LazyImport
+from haystack.tools import Tool, Toolset
+from haystack.utils.auth import Secret
+
+from haystack_experimental.dataclasses.chat_message import ChatMessage, ChatRole, ImageContent, TextContent
+
+with LazyImport("Run 'pip install amazon-bedrock-haystack'") as bedrock_integration_import:
+    import haystack_integrations.components.generators.amazon_bedrock.chat.chat_generator as original_chat_generator
+    import haystack_integrations.components.generators.amazon_bedrock.chat.utils as original_utils
+    from haystack_integrations.components.generators.amazon_bedrock.chat.utils import (
+        _format_tool_call_message,
+        _format_tool_result_message,
+        _repair_tool_result_messages,
+    )
+
+# NOTE: The following implementation ensures that:
+# - we reuse existing code where possible
+# - people can use haystack-experimental without installing amazon-bedrock-haystack.
+#
+#
+# If amazon-bedrock-haystack is installed: all works correctly.
+#
+# If amazon-bedrock-haystack is not installed:
+# - haystack-experimental package works fine (no import errors).
+# - AmazonBedrockChatGenerator fails with ImportError at init (due to bedrock_integration_import.check()).
+
+if not bedrock_integration_import.is_successful():
+    @component
+    class AmazonBedrockChatGenerator():
```

Comment:
Are there extra parentheses? 
```suggestion
    class AmazonBedrockChatGenerator:
```